src/views/RecIM/components/List/Listing/index.jsx

```diff
@@ -356,6 +360,23 @@ const ParticipantListing = ({
     attended ? await updateAttendance(matchID, att) : await removeAttendance(matchID, att);
   };
 
+  const handleChangeParticipantStatus = async () => {
+    if (participant.Status === 'Pending') {
+      let patchedStatus = {
+        StatusID: 4,
+        EndDate: null,
+      };
+      await editParticipantStatus(participant.Username, patchedStatus);
+    } else {
+      let patchedStatus = {
+        StatusID: 1,
+        EndDate: null,
+      };
+      await editParticipantStatus(participant.Username, patchedStatus);
+    }
```

Comment:
```suggestion
    let patchedStatus = {
      StatusID: 1,
      EndDate: null,
    };
    if (participant.Status === 'Pending') {
      patchedStatus.StatusID = 4;
    }
    await editParticipantStatus(participant.Username, patchedStatus);
```
This is a little bit cleaner and simpler IMO.

Reply:
Apparently this did not work, I haven't tested it but it was mentioned when I first saw it. If this works, we could potentially implement an even shorter version similar to 
```
await editParticipantStatus(participant.Username, { StatusID: participant.Status === 'Pending' ? 4 : 1 }
```

Reply:
I didn't want to go crazy on the efficiency :)